### DISCUSSION TITLE
registering multiple components within the same scope

### DISCUSSION THREAD

**Discussion by frutas-fruit** [xissburg/edyn#93] [General]

**Comment by frutas-fruit** [comment]


**Reply by xissburg** [reply]
Yes you should call `registry_external_components` only once listing all components. If you are emplacing or modifying these `RayCast` components in the main registry you'll have to explicitly tell Edyn to update them into the islands by marking them as dirty, e.g.`registry.get_or_emplace<edyn::dirty>(car_id).created<RayCast>();` after emplacing `RayCast`.

**Reply by xissburg** [reply]
Oh you can't have multiple components of the same type in a single entity. You'll have to use one component containing an array.

**Reply by xissburg** [reply]
Oh sorry, I had not noticed. These entities are not associated with a rigid body, thus they're not part of the entity graph. Only entities in the entity graph are shared with the background threads, i.e. between island coordinator and island worker. To make them part of the entity graph you'll have to tag them as external (to turn it into a graph node) and create a null constraint between `ray` and `car_id` (to create an edge in the graph which connects it to a rigid body). See https://github.com/xissburg/edyn/wiki/User-Manual#external-entities

**Reply by xissburg** [reply]
Yes, mark them as dirty in the external system function after changing their values.
